src/nupic/algorithms/AnomalyLikelihood.hpp

```diff
@@ -0,0 +1,261 @@
+    @param timestamp - (optional) timestamp of the ocurrence,
+                       default (-1) results in using iteration step.
+    @return the anomalyLikelihood for this record.
+    **/
+    Real anomalyProbability(Real anomalyScore, int timestamp=-1);
+
+  //public constants:
+  /** "neutral" anomalous value; 
+    * returned at the beginning until the system is burned-in; 
+    * 0.5 (from <0..1>) means "neither anomalous, neither expected"
+    */
+    const Real DEFAULT_ANOMALY = 0.5; 
+    /**
+     * minimal thresholds of standard distribution, if values get lower (rounding err, constant values)
+     * we round to these minimal defaults
+     */
+    const Real THRESHOLD_MEAN = 0.03;
+    const Real THRESHOLD_VARIANCE = 0.0003; 
+
+  private:
+    //methods:
+
+  /**
+  Given a series of anomaly scores, compute the likelihood for each score. This
+  function should be called once on a bunch of historical anomaly scores for an
+  initial estimate of the distribution. It should be called again every so often
+  (say every 50 records) to update the estimate.
+
+  :param anomalyScores: a list of anomaly scores
+
+                        For best results, the list should be between 1000
+                        and 10,000 records
+  :param averagingWindow: integer number of records to average over
+  :param skipRecords: integer specifying number of records to skip when
+                      estimating distributions. If skip records are >=
+                      len(anomalyScores), a very broad distribution is returned
+                      that makes everything pretty likely.
+  :param verbosity: integer controlling extent of printouts for debugging
+
+                      0 = none
+                      1 = occasional information
+                      2 = print every record
+
+  :returns: vector of lilelihoods: , one for each aggregated point
+  **/
+    vector<Real> estimateAnomalyLikelihoods(vector<Real> anomalyScores, UInt skipRecords=0, UInt verbosity=0);
+
+  /**
+  Compute updated probabilities for anomalyScores using the given params.
+
+  :param anomalyScores: a list of records. Each record is a list with the
+
+  :param verbosity: integer controlling extent of printouts for debugging
+  :type verbosity: UInt
+
+  :returns: a vector of likelihoods, one for each aggregated point
+  **/
+    vector<Real>  updateAnomalyLikelihoods(vector<Real> anomalyScores, UInt verbosity=0);
+ /**
+  Given the normal distribution specified by the mean and standard deviation
+  in distributionParams (the distribution is an instance member of the class), 
+  return the probability of getting samples further from the mean. 
+  For values above the mean, this is the probability of getting
+  samples > x and for values below the mean, the probability of getting
+  samples < x. This is the Q-function: the tail probability of the normal distribution.
+  **/
+    Real32 tailProbability(Real32 x) const;
```

Comment:
If you intend this to be a private method, it should have a trailing underscore.